ocaml/xapi/repository_helpers.ml

```diff
@@ -104,13 +104,13 @@ module Pkg = struct
      *   "qemu-dp-2:2.12.0-2.0.11.x86_64", or
      *   "time-(none):1.7-45.el7.x86_64".
      * These may come from "yum updateinfo list updates" and "rpm -qa" *)
-    let r = Re.Str.regexp "^\\(.+\\)\\.\\(noarch\\|x86_64\\)$" in
-    match Re.Str.string_match r s 0 with
-    | true -> (
+    let r = Re.Posix.compile_pat {|^(.+)\.(noarch|x86_64)$|} in
```

Comment:
This could simply split the string from reverse on `.` at most once and then match `("noarch" | "x86_64" as arch)`. No need for a regex here. E.g. `Astring.String.cut ~rev:true` could be used.
No regex needed.